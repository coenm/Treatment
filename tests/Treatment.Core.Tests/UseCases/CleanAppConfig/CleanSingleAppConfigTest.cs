namespace Treatment.Core.Tests.UseCases.CleanAppConfig
{
    using System.IO;
    using System.Threading.Tasks;

    using ApprovalTests;

    using FakeItEasy;

    using Treatment.Core.Interfaces;
    using Treatment.Core.Tests.Resources;
    using Treatment.Core.UseCases.CleanAppConfig;

    using Xunit;

    public class CleanSingleAppConfigTest
    {
        private const string AppConfigFilename = "app.config";
        private readonly IFileSystem fileSystem;
        private readonly CleanSingleAppConfig sut;

        public CleanSingleAppConfigTest()
        {
            fileSystem = A.Fake<IFileSystem>();
            sut = new CleanSingleAppConfig(fileSystem);
        }

        [Fact]
        public async Task ExecuteShouldFixCsProjFileAndDeleteAppconfigFileTest()
        {
            // arrange
            const string csprojFilename = "FileWithRelativeHintPath.txt";
            string outputContent = null;
            A.CallTo(() => fileSystem.OpenRead(csprojFilename, false)).Returns(ResourceFile.OpenRead(csprojFilename));
            A.CallTo(() => fileSystem.SaveContentAsync(csprojFilename, A<Stream>._))
             .Invokes(call =>
                      {
                          var outputStream = call.Arguments[1] as MemoryStream;
                          outputContent = System.Text.Encoding.UTF8.GetString(outputStream?.ToArray() ?? new byte[0]);
                      });

            // act
            await sut.ExecuteAsync(csprojFilename, AppConfigFilename);

            // assert
            A.CallTo(() => fileSystem.DeleteFile(AppConfigFilename)).MustHaveHappenedOnceExactly();
            Approvals.Verify(outputContent);
        }

        [Fact]
        public async Task ExecuteAsyncShouldNotDeleteAppConfigFileWhenCsprojFileDidNotChangeTest()
        {
            // arrange
            const string csprojFilename = "FileWithRelativeHintPathWithoutAppConfig.txt";
            A.CallTo(() => fileSystem.OpenRead(csprojFilename, false)).Returns(ResourceFile.OpenRead(csprojFilename));

            // act
            await sut.ExecuteAsync(csprojFilename, AppConfigFilename);

            // assert
            A.CallTo(() => fileSystem.DeleteFile(AppConfigFilename)).MustNotHaveHappened();
            A.CallTo(() => fileSystem.SaveContentAsync(csprojFilename, A<Stream>._)).MustNotHaveHappened();
        }
    }
}
