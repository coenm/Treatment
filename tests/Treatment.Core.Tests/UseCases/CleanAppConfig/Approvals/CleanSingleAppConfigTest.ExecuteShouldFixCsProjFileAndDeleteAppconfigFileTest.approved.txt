﻿<?xml version="1.0" encoding="utf-8"?>
<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
  <PropertyGroup>
    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
    <Platform Condition=" '$(Platform)' == '' ">x86</Platform>
    <ProjectGuid>{037B7F26-208A-4A0D-8E63-93D3F5E7C86D}</ProjectGuid>
    <OutputType>Library</OutputType>
    <AppDesignerFolder>Properties</AppDesignerFolder>
    <TargetFrameworkVersion>v4.5.2</TargetFrameworkVersion>
    <FileAlignment>512</FileAlignment>
    <OutputPath>bin\$(Configuration)\</OutputPath>
    <PlatformTarget>AnyCPU</PlatformTarget>
  </PropertyGroup>
  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|x64'">
    <DebugSymbols>true</DebugSymbols>
    <DefineConstants>DEBUG;TRACE</DefineConstants>
    <DebugType>full</DebugType>
  </PropertyGroup>
  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release|x64'">
    <DefineConstants>TRACE</DefineConstants>
    <Optimize>true</Optimize>
    <DebugType>pdbonly</DebugType>
  </PropertyGroup>
  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|x86'">
    <DebugSymbols>true</DebugSymbols>
    <DefineConstants>DEBUG;TRACE</DefineConstants>
    <DebugType>full</DebugType>
  </PropertyGroup>
  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release|x86'">
    <DefineConstants>TRACE</DefineConstants>
    <Optimize>true</Optimize>
    <DebugType>pdbonly</DebugType>
  </PropertyGroup>
  <ItemGroup>
    <Reference Include="A.BC, Version=1.0.0.0, Culture=neutral, processorArchitecture=MSIL">
      <HintPath>..\..\..\..\..\..\Packages\A.BC.1.0.149\lib\net45\A.BC.dll</HintPath>
    </Reference>
    <Reference Include="A.BC.External, Version=1.0.0.0, Culture=neutral, processorArchitecture=MSIL">
      <HintPath>..\..\..\..\..\..\Packages\A.BC.1.0.149\lib\net45\A.BC.External.dll</HintPath>
    </Reference>
    <Reference Include="AA.Culture, Version=1.0.0.0, Culture=neutral, processorArchitecture=MSIL">
      <HintPath>$(PackagesDir)\AA.Culture.1.0.2\lib\netstandard1.2\AA.Culture.dll</HintPath>
    </Reference>
    <Reference Include="System" />
    <Reference Include="System.ComponentModel.Composition" />
    <Reference Include="System.Core" />
    <Reference Include="System.Xml.Linq" />
    <Reference Include="System.Data.DataSetExtensions" />
    <Reference Include="Microsoft.CSharp" />
    <Reference Include="System.Data" />
    <Reference Include="System.Net.Http" />
    <Reference Include="System.Xml" />
  </ItemGroup>
  <ItemGroup>
    <None Include="packages.config" />
  </ItemGroup>
  <ItemGroup>
    <ProjectReference Include="..\..\..\DataLayer\DataService\DataService.csproj">
      <Project>{26F6D7EC-B389-4137-81DD-E53FB956AB7A}</Project>
      <Name>DataService</Name>
    </ProjectReference>
  </ItemGroup>
  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
  <Import Project="..\..\..\..\build\XXA.Build.targets" />
</Project>